From: "Jaldhar H. Vyas" <jaldhar@debian.org>
Date: Sat, 7 Sep 2013 01:16:14 -0400
Subject: Update Dovecot name to include Distribution in login greeting
 message

This patch was found at: https://blueprints.launchpad.net/ubuntu/+spec/servercloud-s-server-app-banner-updates
and originally provided by Ubuntu for their 2.1.7 package.  The original
author is  Yolanda Robla <yolanda.robla@canonical.com>.  I updated it
for Debian's 2.2.5 package.

Last-Update: 2020-05-23
---
 configure.ac                         | 7 ++++++-
 src/config/all-settings.c            | 6 +++---
 src/lib-smtp/smtp-server.c           | 2 +-
 src/lmtp/lmtp-settings.c             | 2 +-
 src/login-common/login-settings.c    | 2 +-
 src/submission/submission-settings.c | 2 +-
 6 files changed, 13 insertions(+), 8 deletions(-)

diff --git a/configure.ac b/configure.ac
index 96d1b8c..3479e29 100644
--- a/configure.ac
+++ b/configure.ac
@@ -287,7 +287,12 @@ AM_ICONV
 # SIZE_MAX is missing without this
 CXXFLAGS="$CXXFLAGS -D__STDC_LIMIT_MACROS"
 
-AC_DEFINE_UNQUOTED(DOVECOT_NAME, "$PACKAGE_NAME", [Dovecot name])
+if test -x /usr/bin/lsb_release && lsb_release -si; then
+    DOVECOT_DISTRIBUTION=" ($(lsb_release -si))"
+else
+    DOVECOT_DISTRIBUTION=""
+fi
+AC_DEFINE_UNQUOTED(DOVECOT_NAME, "$PACKAGE_NAME$DOVECOT_DISTRIBUTION", [Dovecot name])
 AC_DEFINE_UNQUOTED(DOVECOT_STRING, "$PACKAGE_STRING", [Dovecot string])
 AC_DEFINE_UNQUOTED(DOVECOT_VERSION, "$PACKAGE_VERSION", [Dovecot version])
 
diff --git a/src/config/all-settings.c b/src/config/all-settings.c
index fc16f83..7724c66 100644
--- a/src/config/all-settings.c
+++ b/src/config/all-settings.c
@@ -2315,7 +2315,7 @@ static const struct submission_settings submission_default_settings = {
 
 	.hostname = "",
 
-	.login_greeting = PACKAGE_NAME" ready.",
+	.login_greeting = DOVECOT_NAME" ready.",
 	.login_trusted_networks = "",
 
 	.recipient_delimiter = "+",
@@ -4111,7 +4111,7 @@ static const struct setting_define login_setting_defines[] = {
 static const struct login_settings login_default_settings = {
 	.login_trusted_networks = "",
 	.login_source_ips = "",
-	.login_greeting = PACKAGE_NAME" ready.",
+	.login_greeting = DOVECOT_NAME" ready.",
 	.login_log_format_elements = "user=<%u> method=%m rip=%r lip=%l mpid=%e %c session=<%{session}>",
 	.login_log_format = "%$: %s",
 	.login_access_sockets = "",
@@ -4322,7 +4322,7 @@ static const struct lmtp_settings lmtp_default_settings = {
 
 	.lmtp_client_workarounds = "",
 
-	.login_greeting = PACKAGE_NAME" ready.",
+	.login_greeting = DOVECOT_NAME" ready.",
 	.login_trusted_networks = "",
 
 	.mail_plugins = "",
diff --git a/src/lib-smtp/smtp-server.c b/src/lib-smtp/smtp-server.c
index f48b60a..3f8cee4 100644
--- a/src/lib-smtp/smtp-server.c
+++ b/src/lib-smtp/smtp-server.c
@@ -47,7 +47,7 @@ struct smtp_server *smtp_server_init(const struct smtp_server_settings *set)
 	if (set->login_greeting != NULL && *set->login_greeting != '\0')
 		server->set.login_greeting = p_strdup(pool, set->login_greeting);
 	else
-		server->set.login_greeting = PACKAGE_NAME" ready.";
+		server->set.login_greeting = DOVECOT_NAME" ready.";
 	if (set->capabilities == 0) {
 		server->set.capabilities = SMTP_SERVER_DEFAULT_CAPABILITIES;
 	} else  {
diff --git a/src/lmtp/lmtp-settings.c b/src/lmtp/lmtp-settings.c
index 5cec2be..8e018f7 100644
--- a/src/lmtp/lmtp-settings.c
+++ b/src/lmtp/lmtp-settings.c
@@ -91,7 +91,7 @@ static const struct lmtp_settings lmtp_default_settings = {
 
 	.lmtp_client_workarounds = "",
 
-	.login_greeting = PACKAGE_NAME" ready.",
+	.login_greeting = DOVECOT_NAME" ready.",
 	.login_trusted_networks = "",
 
 	.mail_plugins = "",
diff --git a/src/login-common/login-settings.c b/src/login-common/login-settings.c
index b1e0b06..f133717 100644
--- a/src/login-common/login-settings.c
+++ b/src/login-common/login-settings.c
@@ -50,7 +50,7 @@ static const struct setting_define login_setting_defines[] = {
 static const struct login_settings login_default_settings = {
 	.login_trusted_networks = "",
 	.login_source_ips = "",
-	.login_greeting = PACKAGE_NAME" ready.",
+	.login_greeting = DOVECOT_NAME" ready.",
 	.login_log_format_elements = "user=<%u> method=%m rip=%r lip=%l mpid=%e %c session=<%{session}>",
 	.login_log_format = "%$: %s",
 	.login_access_sockets = "",
diff --git a/src/submission/submission-settings.c b/src/submission/submission-settings.c
index c4dd3db..5b3e00d 100644
--- a/src/submission/submission-settings.c
+++ b/src/submission/submission-settings.c
@@ -103,7 +103,7 @@ static const struct submission_settings submission_default_settings = {
 
 	.hostname = "",
 
-	.login_greeting = PACKAGE_NAME" ready.",
+	.login_greeting = DOVECOT_NAME" ready.",
 	.login_trusted_networks = "",
 
 	.recipient_delimiter = "+",
